geopandas/array.py

```diff
@@ -828,14 +828,27 @@ def estimate_utm_crs(self, datum_name="WGS 84"):
             raise RuntimeError("crs must be set to estimate UTM CRS.")
 
         minx, miny, maxx, maxy = self.total_bounds
+        x_center = None
+        y_center = None
         # ensure using geographic coordinates
         if not self.crs.is_geographic:
-            lon, lat = Transformer.from_crs(
-                self.crs, "EPSG:4326", always_xy=True
-            ).transform((minx, maxx, minx, maxx), (miny, miny, maxy, maxy))
-            x_center = np.mean(lon)
-            y_center = np.mean(lat)
-        else:
+            transformer = Transformer.from_crs(self.crs, "EPSG:4326", always_xy=True)
+            if compat.PYPROJ_GE_31:
+                minx, miny, maxx, maxy = transformer.transform_bounds(
+                    minx, miny, maxx, maxy
+                )
+                if minx > maxx:
+                    # crossed the antimeridian
+                    minx = -180
+                    maxx = -180
+            else:
```

Comment:
Can you adapt one of the CI envs to ensure this is tested?

Reply:
It seems [37-minimal.yml](https://github.com/geopandas/geopandas/blob/8547d7fff6522a3499517d35f00c5751d2f05330/ci/envs/37-minimal.yaml#L28) is not working as expected?

Reply:
Seems to pull in an earlier version of pyproj [ref](https://github.com/geopandas/geopandas/pull/2049/checks?check_run_id=3276059699#step:4:44).

Reply:
Ah, wee need `pyproj>=3.0.0,<3.1.0`. Do you have a preference for where this pin is placed?

Reply:
An alternative would be to require pyproj 3.1 in this method so the behavior is consistent.

Reply:
I think you can update the pin in `37-pd10.yaml`.